ocs_ci/helpers/dr_helpers_ui.py

```diff
@@ -392,3 +390,93 @@ def verify_failover_relocate_status_ui(
         log.info("Close button found")
         acm_obj.do_click_by_xpath("//*[text()='Close']")
         log.info("Data policy modal page closed")
+
+
+def verify_application_not_present_in_ui(acm_obj, workload_to_check, timeout=60):
+    """
+    Verify if application is not present in UI
+
+    Args:
+        acm_obj (AcmAddClusters): ACM Page Navigator Class
+        workload_to_check (str): Specify the workload to check,
+        timeout (int): timeout to wait for certain elements to be found on the ACM UI
+
+    Returns:
+        bool: True if the application is not present, raises Exception if multiple matches found
+
+    """
+    if workload_to_check:
+        ocp_version = get_ocp_version()
+        acm_loc = locators[ocp_version]["acm_page"]
+        acm_obj.navigate_applications_page()
+        log.info("Click on search bar")
+        acm_obj.do_click(acm_loc["search-bar"])
+        log.info("Clear existing text from search bar if any")
+        acm_obj.do_clear(acm_loc["search-bar"])
+        log.info("Enter the workload to be searched")
+        acm_obj.do_send_keys(acm_loc["search-bar"], text=workload_to_check)
+        action_status = acm_obj.wait_until_expected_text_is_found(
+            locator=acm_loc["no-results-found"],
+            expected_text="No results found",
+            timeout=timeout,
+        )
+        if action_status:
+            fetch_status = acm_obj.get_element_text(locator=acm_loc["no-results-found"])
+            assert action_status, "Application present in UI"
+            log.info(
+                f"Successfully verified on ACM UI, Application Not Found, fetch status is {fetch_status}"
+            )
+            return True
+        elif not action_status:
+            acm_obj.do_clear(acm_loc["search-bar"])
+            assert acm_obj.do_send_keys(acm_loc["search-bar"], text=workload_to_check)
+            return True
+        else:
+            raise InvalidSelectorException(
+                "One or more matches found, Incorrect workload name"
+            )
```

Comment:
When it could/should reach to this part, while (I think) the previous conditions `if action_status` and `elif not action_status` covers all possible cases (`action_status` evaluated to `True` or `False`)?

Reply:
As discussed over the call, elif and else conditions should be handled properly. We aren't doing any checks as part of these loops.

Reply:
The check part is modified, Please check